v6.0.0-beta.1 (#7727)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.0.0-beta.1

_Jan 27, 2023_

We'd like to offer a big thanks to the 14 contributors who made this release possible. Here are some highlights ✨:

> - 🌍 Add Belarusian (be-BY), Czech (cs-CZ) and Russian (ru-RU) locales

Reviewer: ```suggestion
- 🚀 New shortcuts component for the date pickers
- 🌍 Add Belarusian (be-BY), Czech (cs-CZ) and Russian (ru-RU) locales
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.0.0-beta.1

_Jan 27, 2023_

> We'd like to offer a big thanks to the 14 contributors who made this release possible. Here are some highlights ✨:

Reviewer 2: 🤩 🎉 🚀 
```suggestion
We'd like to offer a big thanks to the 16 contributors who made this release possible. Here are some highlights ✨:
```

- 🚀 New shortcuts component for the date pickers (#7154) @alexfauquette
- 🌍 Add Belarusian (be-BY), Czech (cs-CZ) and Russian (ru-RU) locales

> - 🌍 Improve Spanish (es-ES), Japanese (ja-JP) and Slovak (sk-SK) locales

Author: If I am not mistaken Belarusian (be-BY) is [newly added](https://github.com/mui/mui-x/pull/7646) in this release, shouldn't we mention it as `Add Belarusian (be-BY) ...` 🤔

> - 🌍 Improve Spanish (es-ES), Japanese (ja-JP) and Slovak (sk-SK) locales

Reviewer 2: Damn. Fair point. 🙈 
Need to check if it's not a mistake on v5 release suggestion... :/

- ✨ New codemods for migrating to v6
- 📚 Documentation improvements
- 🐞 Bug fixes

### `@mui/x-data-grid@v6.0.0-beta.1` / `@mui/x-data-grid-pro@v6.0.0-beta.1` / `@mui/x-data-grid-premium@v6.0.0-beta.1`

#### Changes

> - [DataGrid] Add `title` to cells (#7682) @thupi

Reviewer 3: ```suggestion
- [DataGrid] Add `title` attribute to cells (#7682) @thupi
```

Otherwise it's tricky to know it it's a props we added, or a native behavior

Author: Sure, I'll update the same in [v5 release](https://github.com/mui/mui-x/pull/7726/files) too

- [DataGrid] Fix `autoHeight` not working properly inside of a flex container (#7701) @cherniavskii
- [DataGrid] Fix grid state not being updated after print preview is closed (#7642) @cherniavskii
- [DataGrid] Fix non-hideable columns visibility toggling (#7637) @cherniavskii
- [DataGrid] Fix scrolling on resize for data grids inside shadow root (#7298) @akiradev
- [l10n] Add Slovak (sk-SK) translation for aggregation functions (#7702) @m4theushw
- [l10n] Add missing core locales for `MuiTablePagination` (#7717) @MBilalShafi

> - [l10n] Improve Spanish (es-ES) and Vietnamese (vi-VN) locale (from @WiXSL and @SpacerZ) (#7634) @alexfauquette

Reviewer 3: ```suggestion
- [l10n] Improve Spanish (es-ES) and Vietnamese (vi-VN) locale (#7634) @WiXSL and @SpacerZ
```

- [l10n] Add Belarusian (be-BY) locale (#7646) @volhalink

### `@mui/x-date-pickers@v6.0.0-beta.1` / `@mui/x-date-pickers-pro@v6.0.0-beta.1`

#### Changes

- [pickers] Fix `aria-labelledby` assignment to dialog (#7608) @LukasTy
- [pickers] Support `UTC` with `dayjs` (#7610) @flaviendelangle
- [pickers] Update focus when opening a UI view (#7620) @alexfauquette
- [DateRangePickers] Add shortcuts component (#7154) @alexfauquette
- [l10n] Add Czech (cs-CZ) locale (#7645) @OndrejHj04

> - [l10n] Add Russian (ru-RU) locale  (#7706) @rstmzh

Reviewer 3: ```suggestion
- [l10n] Add Russian (ru-RU) locale (#7706) @rstmzh
```

> - [l10n] Improve Japanese (ja-JP) locale (#7624) @makoto14

Author: This contains locale updates for both grid and pickers. Should I move it to `[core]` section or it's fine in either of grid or pickers?
Did we have similar cases previously?

Reviewer 2: Hmm, maybe just list it in both data-grid as well as pickers changes? 🤔

Reviewer 2: Other than this and optionally the inclusion of v5 changelog after it's release, LGTM. 👌

Author: Or should we think about having a dedicated section for `Localization/l10n` ?

Reviewer 2: Indeed, that could also work 👍 
But these few releases do seem like a bit of an abnormal case with very huge amount of localization contributions.
Usually it's way more quite on this regard 🙈

Author: It's not that big of an issue, as more contribution percentage is on the pickers side I'll keep it here for this PR.

### `@mui/x-codemod@v6.0.0-beta.1`

#### Changes

> - [codemod] Update pickers component toolbar props (#7687) @alexfauquette

Reviewer 2: ```suggestion
- [codemod] Add pickers `replace-toolbar-props-by-slot` codemod (#7687) @alexfauquette
```

> - [codemod]  Update pickers component toolbar props (#7687) @alexfauquette

Author: ```suggestion
- [codemod] Update pickers component toolbar props (#7687) @alexfauquette
```

- [codemod] Add `GridColumnMenuItemProps` to `column-menu-components-rename` codemod (#7710) @MBilalShafi
- [codemod] Add `headerHeight` prop update to `row-selection-props-rename` codemod (#7711) @MBilalShafi

> - [codemod] Add codemod for `components` to `slots` remaining (#7533) @alexfauquette

Reviewer 2: ```suggestion
- [codemod] Add pickers codemod for `components` to `slots` renaming (#7533) @alexfauquette
```

> - [codemod] Basics migration to components props (#7698) @alexfauquette

Reviewer 2: ```suggestion
- [codemod] Add pickers `migrate-to-components-componentsProps` and `replace-arrows-button-slot` codemods (#7698) @alexfauquette
```

> - [codemod] Rename `rowsPerPageOptions` prop to `pageSizeOptions` (#7603) @MBilalShafi

Reviewer 2: ```suggestion
- [codemod] Add data grid codemod renaming `rowsPerPageOptions` prop to `pageSizeOptions` (#7603) @MBilalShafi
```

> - [codemod] Rename `shouldDisableTime` prop (#7709) @alexfauquette

Reviewer 2: ```suggestion
- [codemod] Add pickers `rename-should-disable-time` codemod (#7709) @alexfauquette
```

> - [codemod] Rename data grid props (#7485) @MBilalShafi

Reviewer 2: ```suggestion
- [codemod] Add data grid `row-selection-props-rename` codemod (#7485) @MBilalShafi
```

> - [codemod] Rename data grid selectors and events (#7699) @MBilalShafi

Reviewer 2: ```suggestion
- [codemod] Add data grid `rename-selectors-and-events` codemod (#7699) @MBilalShafi
```

> - [codemod] Update `DateTimePicker` component `tabs` props (#7639) @alexfauquette

Reviewer 2: ```suggestion
- [codemod] Add pickers `replace-tabs-props` codemod (#7639) @alexfauquette
```
